server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/DownloadRoutes.scala

```diff
@@ -297,14 +298,14 @@ class DownloadRoutes @Inject()(@Named(InjectionKeys.RFC_8621) val authenticator:
     Id.validate(request.param(accountIdParam)) match {
       case Right(id: Id) =>
         val targetAccountId: AccountId = AccountId(id)
-        AccountId.from(mailboxSession.getUser).map(accountId => accountId.equals(targetAccountId))
-          .fold[SMono[Unit]](
-            e => SMono.error(e),
-            value => if (value) {
-              get(request, response, mailboxSession)
-            } else {
-              SMono.error(ForbiddenException())
-            })
+
+        sessionTranslator.delegateIfNeeded(mailboxSession, targetAccountId)
+          .flatMap(session => AccountId.from(session.getUser)
+            .fold(e => SMono.error(e),
+              accountId => SMono.just(accountId.equals(targetAccountId))
+                .filter(containAccountId => containAccountId)
+                .flatMap(_ => get(request, response, session))
+                .switchIfEmpty(SMono.error(ForbiddenException()))))
```

Comment:
This code is hard to read I am pretty sure it can be refactored to be more readable...

I am pretty sure the `(accountId.equals(targetAccountId)` check is now useless and can be removed:

```
sessionTranslator.delegateIfNeeded(mailboxSession, targetAccountId)
    .flatMap(session => get(request, response, session))
```

Reply:
It really pretty